### DISCUSSION TITLE
Increase count by one when tapping

### DISCUSSION THREAD

**Discussion by sokai** [iSoron/uhabits#994] [Feature Requests]
Is it possible to have a numeric counter/measurable habit that, when it is tapped, counts 1 (or maybe possible to adjust) up?

FYI: In combination with https://github.com/iSoron/uhabits/discussions/993 (recording time stamps) it would be the *same* behaviour, like [Better counter](https://github.com/albertvaka/bettercounter) works.

Thanks and KR!

**Comment by VA-MMXXII** [comment]
I totally agree. The numeric counter should 1) offer an integer option and 2) it should have a simple widget with + on one side and - on the other.

Alongside Loop I have also been using Daily Counter (https://play.google.com/store/apps/details?id=com.turquoisetools.dailycounter&hl=en_US&gl=US) for years and it has exactly such a widget. But the app hasn't been updated since March 12, 2020 after a previously good track record of updates and bug fixes and the developer's Website is also down and he hasn't been returning e-mails for over a year after previously being very responsive. So I would love to have a resetting daily or weekly counter solution right inside Loop, with appropriate reporting, too.

**Comment by YalandHong** [comment]
Related to #951

**Comment by soshial** [comment]
What about this solution? Simply adding a "Plus" button alongside the value dialog.

<img width="400px" src="https://user-images.githubusercontent.com/100644/219103411-98c68ede-945f-459e-b91d-c25803308008.jpg"/>

**Reply by VA-MMXXII** [reply]
Makes sense with the current interface, but 1) it'd be good to also have a "-" button, in case there is an error or change, and 2) for whatever reason, when entered from a widget, counts in the current version of Loop Habits also open up the app, which is not desirable (oftentimes the whole point of a widget on a home screen is to quickly enter something w/o opening the app). What I was really talking about was a widget enhancement or option that looks something like the attached counter widgets (from the old Daily Counter app).
![counter_widget](https://user-images.githubusercontent.com/92600518/219111109-49871b2d-81c8-40df-85aa-c4ee4b2314e6.png)

**Reply by herrbrixter** [reply]
I agree with this proposal.

Also please allow decimals  I'd like to increment by 0.5. Use case is for drinking 3 liters of water per day where 1 mug is 500ml.

0, 0.5, 1, 1.5, 2, 2.5, 3

**Comment by derrowap** [comment]
Came here to make a similar request. Water consumption is one thing I like to track and instead of adding Xoz to the existing amount in my head multiple times a day (and typing it out), I was hoping to have a button like "+X" where X is programmable on a per-habit basis. Since I typically use the same cup, this would reduce the resistance of tracking this. Could be it's own widget, but better yet could be in the current widget's popup window ("Change value") below the area you type the current value in parallel with the "ok" button.

**Comment by cbundy** [comment]
This is my one frustration from the app so far and it's driving me to add more binary habits than measurable ones just to make it easier.

I agree with above, tap to increase be 1 should be an option.

If you take it further, perhaps the goal number could help prompt the widget behaviour.

If my goal is 2 for the day, tap or whatever to increase by 1

If my goal for the week is 100, maybe it opens the widget and I see the number and a +1 and +5 or +10 button as well.

**Comment by nutpantz** [comment]
I also want this option both + and - 
BUT I would also like an option to set the incremental value
So in settings I can set the + to increase it by 5 or 10 (or whatever) for each tap.
(It would be cool it I could set the minus as a separate value or keep it at 1 but not at all necessary)
I don't need it setable by habit.
Going to 100 by 5's I can drop by 1's to get an exact number like 147 very fast

**Comment by aaehub** [comment]
Just make it increase 1 by taping, i hate entering the app every time just to increase by one. Pain

**Comment by nisc** [comment]
I would also appreciate this feature. I have several items to track "How often did I do X today?", and currently it's very cumbersome to do this, given there are often 20+ occurrences per day.

**Comment by awisync** [comment]
Just add a + and - button. Kindly improve the app step by step.

**Comment by gpukeko** [comment]
Agree with the other suggestions here. Would love a feature like this. With the binary habits, long press to check is very convenient. Long press to increase count by one would be very useful. I have 14 habits and 11 of them are numeric, with my goal either twice or three times per day.

**Comment by Chepycou** [comment]
If the idea is not to clutter the interface too much maybe a three-fold input oculd be :
- one-tap : open the keyboard like it is now
- long-press then swipe up : increase (the amount can be set in the settings)
- long-press then swipe down : decrease (the amount can also be set in the settings)

This way, there is no new dialog and/or new User Interaction method of input
